docs/migration-guides/prometheus-migration.md

```diff
@@ -0,0 +1,351 @@
+
+This document describes the steps to migrate the TSDB from an existing Prometheus deployment to the one installed by BKPR.
+
+Furthermore, this process will not generate any downtime for your current Prometheus deployment, making it safe to be followed.
+
+## Pre-requisites
+
+The following prerequisites should be met before starting with the migration process:
+
+* You have a healthy Kubernetes cluster running.
+* You have a Prometheus deployment that *matches the BKPR Prometheus version*, which can be obtained from [the BKPR release compatibility table](https://github.com/bitnami/kube-prod-runtime#release-compatibility).
+* You have already read the [Kubeprod workflow documentation](https://github.com/bitnami/kube-prod-runtime/blob/a304cbc5b12042f2a17f96f23cd62a9759e81b4b/docs/workflow.md)
+* [Kubecfg binary](https://github.com/ksonnet/kubecfg/releases)
+
+This article will not cover the Proemtheus upgrade process. If you need to update your Prometheus deployment, check the [Prometheus docs](https://prometheus.io/docs/prometheus/latest/migration/)
+
+## Overview
+
+This section provides an overview of the required steps to migrate the Prometheus TSDB from an already running deployment to the BKPR Prometheus one.
+
+This process has been designed and tested to ensure the shortest possible downtime when migrating to the new StatefulSet by having two Prometheus instances running at the same time. These instances will be modified to include a sidecar container that will deploy an *rsync* client and server with access to the Prometheus data volume in order to move a Prometheus TSDB snapshot from one pod to another.
+
+Before starting the migration it is recommended to get familiar with the concepts of TSDB and Write-Ahead-Log (WAL) file. See the following links for more information:
+
+* [Prometheus TSDB](https://prometheus.io/docs/prometheus/latest/storage/)
+
+* [Write-Ahead-Log file](https://github.com/prometheus/tsdb/blob/master/docs/format/wal.md)
+
+As Prometheus doesn’t yet include a *flush* method to dump the WAL file that is being written into a snapshot with a block format that doesn’t overlap time ranges (see [this issue](https://github.com/prometheus/tsdb/issues/346) for more context) it is necessary to sync with the Prometheus TSDB compaction times in order to ensure minimal data loss (less than 5 minutes).
+
+These are the steps required to perform this migration:
+
+* Deploy an *rsync* server-side container in your current Prometheus deployment
+* Modify the `kubeprod-manifest.jsonnet` file to add another sidecar *rsync* container (acting as a client)
+* Sync with the compaction times of the Prometheus database
+* Create a TSDB snapshot and restore it in the new deployment
+* Check next TSDB compacts
+
+## Step 1: Deploy the sidecar *rsync* server-side container
+
+In order to access the Prometheus TSDB snapshots using *rsnyc*, it will be necessary to deploy an *rsync* server with a minimal configuration.
```

Comment:
```suggestion
In order to access the Prometheus TSDB snapshots using *rsync*, it will be necessary to deploy an *rsync* server with a minimal configuration.
```